gl_test/action.yml

```diff
@@ -59,7 +59,9 @@ runs:
         cp tt_submission/*.v "${{ inputs.test-dir }}/gate_level_netlist.v"
         cd "${{ inputs.test-dir }}"
         make clean
+        rm -f results.xml
```

Comment:
The file shouldn't exist at the first place in gl tests? Unless someone committed it to the repo, of course. In that case, maybe it makes better sense to fail rather than delete it?

Reply:
It is present in my actions because some of us don't run a SINGLE GL_TEST iteration,  testcases of more complex projects have multiple modes of operation, so the "make" needs to be run for each mode.
For example I2C is testing in OpenDrain mode and then again in PushPull mode.
For example USB is testing in LowSpeed mode and then again in FullSpeed mode.
For example startup initialization maybe tested with all-zero, all-one and all-random (for flip-flop initial state seeding)

It costs and breaks nothing to have the line present, and provides assurances (when trying to diagnose a problem) the "make" did create the file, etc...

Maybe that is a good one to check contents of .gitignore re accidental commit to repo.

Reply:
Looks like `make clean` already got us covered. The file is deleted not once, but twice (!) 

![image](https://github.com/TinyTapeout/tt-gds-action/assets/892318/1df69802-130b-4ce2-9d1d-9bddf09d0eb1)

Reply:
Hmm that does not satisfy the assurance goals for the purpose of CI (IMHO) where a human is not going to read every line of logfile output.
For example a failure of "make" to do anything but it still returns with success status, now because there is previous history of how cocotb works (your tests can fail but make looks successful, which is Makefile culture anti-pattern, you need to defend your CI against this known issue to make sure what is important is done with assurance).
For the purpose of CI the make tool exit status can not be trusted for any purpose than to report some kinds of failures.  So you can not trust its execution over what it does, if it happened, etc...
So in a scenario where cocotb does not run, this CI defensive measure makes an assurance guarantee (for the purpose of CI exit status continuity) that no matter what `make` is doing, no matter what changes upstream cocotb introduced, the thing that is important to this CI task is results.xml was regenerated.

Just because the file deletion happens via some other mechanism when the task follows the happy path, does not say anything about the non-happy path.  If you leave the extra file removal command in, you remove a headache when performing diagnosic, as you have assurance from tools that can be trusted over what happened.  You did not delegate that task to a `make` run with dubious exist status behaviour.

What is important is the GHA task green light should never show success in scenarios where success did not occur.

An alternative would be to put the "rm -f results.xml" in the `test/Makefile` with the target of `clean` in the TT Makefile that will achieve the same purpose.

Reply:
Thanks for explaining! added it (see the last commit)